reconcile/signalfx_endpoint_monitoring.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from typing import Any, Optional
+
+from reconcile.closedbox_endpoint_monitoring_base import (
+    EndpointMonitoringProvider,
+    Endpoint,
+    parse_prober_url,
+    run_for_provider,
+)
+from reconcile.utils.openshift_resource import OpenshiftResource
+from reconcile.utils.semver_helper import make_semver
+
+QONTRACT_INTEGRATION = "signalfx-prometheus-endpoint-monitoring"
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+LOG = logging.getLogger(__name__)
+
+PROVIDER = "signalfx"
+
+LOG = logging.getLogger(__name__)
+
+
+def run(
+    dry_run: bool, thread_pool_size: int, internal: bool, use_jump_host: bool
+) -> None:
+    run_for_provider(
+        PROVIDER,
+        build_probe,
+        QONTRACT_INTEGRATION,
+        QONTRACT_INTEGRATION_VERSION,
+        dry_run,
+        thread_pool_size,
+        internal,
+        use_jump_host,
+    )
+
+
+def build_probe(
+    provider: EndpointMonitoringProvider, endpoints: list[Endpoint]
+) -> Optional[OpenshiftResource]:
+    signalfx = provider.signalFx
+    if signalfx:
```

Comment:
nit: walrus code golf :golfing: 

```
if signalfx := provider.signalFx:
```

Or better to avoid extra indent completely:

```
if not signalfx:
    return None
prober_url = parse_prober_url(signalfx.exporterUrl)
...
```